pkgs/by-name/ty/typst/typst-packages.nix

```
@@ -0,0 +1,52 @@
+{
+  lib,
+  callPackage,
+}:
+
+lib.makeExtensible (
+  final:
+  lib.recurseIntoAttrs (
+    let
+      toPackageName = name: version: "${name}_${lib.replaceStrings [ "." ] [ "_" ] version}";
+    in
+    lib.foldlAttrs (
+      packageSet: pname: versionSet:
+      packageSet
+      // (lib.foldlAttrs (
+        subPackageSet: version: packageSpec:
+        subPackageSet
+        // {
+          ${toPackageName pname version} = callPackage (
+            {
+              lib,
+              buildTypstPackage,
+              fetchzip,
+            }:
+            buildTypstPackage (finalAttrs: {
+              inherit pname version;
+
+              src = fetchzip {
+                inherit (packageSpec) hash;
+                url = "https://packages.typst.org/preview/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
+                stripRoot = false;
+              };
+
+              typstDeps = builtins.filter (x: x != null) (
+                lib.map (d: (lib.attrsets.attrByPath [ d ] null final)) packageSpec.typstDeps
+              );
+
+              meta = {
+                inherit (packageSpec) description;
+                maintainers = with lib.maintainers; [ cherrypiejam ];
+                license = lib.map (lib.flip lib.getAttr lib.licensesSpdx) packageSpec.license;
+              } // (if packageSpec ? "homepage" then { inherit (packageSpec) homepage; } else { });
+            })
+          ) { };
```

This should make the `<pkg>.typstDeps` attributes change when `typstPackags` get extended.

```suggestion
          ) { typstPackages = final; };
```